app/workers/delete_old_bets_worker.rb

```diff
@@ -0,0 +1,40 @@
+class DeleteOldBetsWorker
+  # All expired bets older than this, or all disabled bets last
+  # updated longer ago than this, are deleted
+  OLD_BET_THRESHOLD = 90.days
+
+  include Sidekiq::Worker
+
+  def perform
+    queries = old_expired_bets.map(&:query_id).uniq + old_disabled_bets.map(&:query_id).uniq
+
+    old_expired_bets.delete_all
+    old_disabled_bets.delete_all
+
+    # remove empty queries from search-api to avoid cruft in the index
+    queries.each do |qid|
+      q = Query.find(qid)
+      if q.bets.empty?
+        SearchApiSaver.new(q).destroy
```

Comment:
Could we perhaps have this as a `before_destroy` step on the models? 

I wonder whether we could have queries be deleted once all of their bets are deleted with an ActiveRecord association?

Reply:
I've added it as a `before_destroy` step.  I'm not sure about doing this through an ActiveRecord association, as a query with no bets is a reasonable thing to have: it's a new query with no bets yet.